src/ImageRunModal.jsx

```diff
@@ -321,6 +384,37 @@ export class ImageRunModal extends React.Component {
         const { image } = this.props;
         const dialogValues = this.state;
 
+        const networkNsMode = (
+            <FormSelect id='run-image-dialog-nsmode'
+                       value={this.state.nsMode}
+                       onChange={value => this.onValueChanged('nsMode', value)}>
+                {
+                    (this.state.image.isSystem
+                        ? ['bridge', 'none', 'host']
+                        : ['none']).map((nsmode) =>
+                        (
+                            <FormSelectOption value={nsmode} key={nsmode} label={nsmode} />
+                        )
+                    )
+                }
+                {
+                    this.state.slirp4netnsVersion
+                        ? (
+                            <FormSelectOption value="slirp4netns" key="slirp4netns" label="slirp4netns" />
+                        )
+                        : null
+                }
+            </FormSelect>
+        );
+        const cniSelector = (
+            <DynamicListForm id='run-image-dialog-cni'
+                             formclass='cni-form'
+                             onChange={value => this.onValueChanged('useCniNetworks', value)}
```

Comment:
Podman API is not 1:1 mapping to `--network` option, it implements API interface with two options `netns.nsmode` and `cni_networks`.

In Podman CLI, we can use `--network <network-id>[,<network-id> ...]` to attach container to user-defined CNI network(s), and in API way, we use `{ netns: { nsmode: "bridge" }, cni_networks: [ "network-id", "network-id", ... ] }`

If I picked `bridge` mode with `foobar`, the CLI option would be `--network=foobar`, but with `foobar` and `abcd` CNIs, it would be `--network=foobar,abcd`. So CNI selector can be multiple choice.

If I choose `bridge` mode without selecting CNIs (`--network=bridge` in CLI), podman will pick the default network (`podman` CNI on my machine) as the only network, and this behavior is as same as `--network=private` (the default network option) now.

I just keep the design in podman API form, I think we can add some tips to telling users that without picking CNI will use default network. Or you think that we should follow the CLI form and create new option for picking CNIs?